### DISCUSSION TITLE
How to include original message to a reply email

### DISCUSSION THREAD

**Discussion by quebecblogue** [PHPMailer/PHPMailer#2568] [Q&A]
Hi,

I am building a custom email support system and I am using phpmailer to build the email... I have tried to Google the solutions for the last few days and either I am using the wrong keywords, the solution is not out there or I am missing something that should be crystal clear.

When replying to an email, I am setting proper In-Reply-To and References header like it is specified in the RFC but I am also trying to include the conversations below my answer like pretty much every email client works.

How do I proceed? Is it something the email server is supposed to do automatically but mine doesn't? Do I need to attach the email as an attachment? Do I only need to append the previous email with proper formatting? And if so, what formatting? Because I tried to mimic Gmail, Hotmail and Yahoo and they all do it differently...

Thanks a lot in advance for your help, It is greatly appreciated.

Cheers.

**Comment by Synchro** [comment]
It really has nothing to do with the mail server, it's entirely down to the client, and it varies a lot. The definitive guide to email threading is probably [this](https://cr.yp.to/immhf/thread.html); it's very old, but so is email!

**Comment by quebecblogue** [comment]
My emails are threading properly.... What I want is to include the trail of
the past messages inside the content of the email.

Like this...

![Screenshot 2021-11-26 11 16 11](https://user-images.githubusercontent.com/14190910/143610994-12b0025f-070f-4362-b720-0a6637559a7f.png)

Best regards,

**Reply by Synchro** [reply]
For some reason I can't see that image. Can you put it directly in this discussion instead of in imgur? There is no particular standard for quoting that I know of other than using multiple levels of `>` and indenting. It varies a lot between clients as a result.

**Reply by quebecblogue** [reply]
I have included the screenshot. Sorry for that.

I guess that would be fairly easy to add quoting to plain text message with multiple levels of > but if I am using multiple levels of > in HTML email, I'll break everything no?

**Reply by Synchro** [reply]
Yes - it's tricky! If you look at the awful code that outlook produces, and the rather better code that Apple Mail generates you might get some ideas, but it's not simple. PHPMailer doesn't really care - it just sends whatever you give it, so this bit is mostly up to you.

Some apps might rely on MIME structure rather than HTML, just having a whole stack of `multipart/mixed` parts instead. PHPMailer lacks support for arbitrary MIME structures, so if you want to do that you might be better off with symfony/mail or zend_mail.

**Reply by quebecblogue** [reply]
Hi Marcus... I just want to tell you that thanks to your enlightenment, I've solved my problem. It was fairly easy once I understood that it was only blockquoting.

If it can help somebody, here is how I did it.

` $message  .= "<br><div class=\"gmail_quote\"><div dir=\"ltr\" class=\"gmail_attr\">".strftime("Le %a %e %B %Y à %H:%M", strtotime($headers["Date"]))." {$nom} &lt;<a href=\"mailto:{$to}\">{$to}</a>&gt; a écrit :</div><blockquote class=\"gmail_quote\" style=\"margin:0 0 0 .8ex;border-left:1px #ccc solid;padding-left:1ex\">{$past_threads}</blockquote></div>";
`

Cheers.
